File: benchmarks/Csharp/Benchmarks/Richards/IdleTaskDataRecord.cs
namespace AreWeFastYet;

internal sealed class IdleTaskDataRecord : RBObject
{
    public int Control { get; set; }

    public int Count { get; set; }

    internal IdleTaskDataRecord()
    {
        Control = 1;
        Count = 10000;
    }
}

File: benchmarks/Rust/src/richards/idle_task_data_record.rs
use crate::richards::rb_object::RBObject;

pub struct IdleTaskDataRecord {
    control: usize,
    count: usize,
}

impl Default for IdleTaskDataRecord {
    fn default() -> Self {
        Self {
            control: 1,
            count: 10000,
        }
    }
}

impl RBObject for IdleTaskDataRecord {}

impl IdleTaskDataRecord {
    pub fn get_control(&self) -> usize {
        self.control
    }

    pub fn set_control(&mut self, control: usize) {
        self.control = control;
    }

    pub fn get_count(&self) -> usize {
        self.count
    }

    pub fn set_count(&mut self, count: usize) {
        self.count = count;
    }
}
